libcoap/src/error.rs

```diff
@@ -82,16 +83,25 @@ pub enum OptionValueError {
     /// A string value could not be converted to UTF-8.
     #[error("CoAP option has invalid value: invalid string")]
     StringConversion(#[from] FromUtf8Error),
+    /// A string value could not be converted to UTF-8.
+    #[error("CoAP option has invalid value: invalid URI")]
+    UriParsing(#[from] UriParsingError),
     /// Option has an illegal value.
     #[error("CoAP option has invalid value")]
     IllegalValue,
 }
 
-#[derive(Error, Debug, Copy, Clone, Eq, PartialEq)]
+#[derive(Error, Debug, Clone, Eq, PartialEq)]
 pub enum UriParsingError {
+    /// Unknown error inside of libcoap
+    #[error("CoAP option creation error: unknown error in call to libcoap")]
+    Unknown,
     /// URI does not have a valid scheme for libcoap (coap, coaps, coap+tcp, coaps+tcp, http, https).
-    #[error("URL does not have scheme valid for libcoap")]
-    NotACoapScheme,
+    #[error("URI scheme {} is not a valid CoAP scheme known to libcoap", .0)]
+    NotACoapScheme(String),
+    /// Provided URI contains a null byte.
+    #[error("Provided URI contains a null byte")]
+    ContainsNulByte(#[from] NulError),
```

Comment:
Is this a misspelling or on purpose (i.e., from the way it's called in ASCII)? Just making sure here.

Reply:
I initially matched the spelling of the error that is encapsulated by this variant, but i think I'll rename it to `ContainsNullByte`.